docs/upgrade.html

```diff
@@ -58,6 +58,11 @@
         whether there is offline log directory. </li>
     <li>Added KafkaStorageException which is a retriable exception. KafkaStorageException will be converted to NotLeaderForPartitionException in the response
         if the version of client's FetchRequest or ProducerRequest does not support KafkaStorageException. </li>
+    <li>The overridden <code>handleError</code> method implementations have been removed from the following deprecated classes from
+        the <code>kafka.api</code> package: <code>FetchRequest</code>, <code>GroupCoordinatorRequest</code>, <code>OffsetCommitRequest</code>,
+        <code>OffsetFetchRequest</code>, <code>OffsetRequest</code>, <code>ProducerRequest</code>, and <code>TopicMetadataRequest</code>.
+	This was only intended for use on the broker, but it no longer in use and the implementations have not been maintained.
+	A stub implementation has been retained for binary compatibility.</li>
```

Comment:
A couple of nits: 
- there are two "from" in near proximity, maybe use `in` for the second one? 
- "but it no" -> "but it is no"